### PR TITLE
Enable listing serverless databases in the explorer

### PR DESCRIPTION
**Pull request by abrooksv**
* Also conditionally disable the IAM auth action

## Issue
#2403 

## Note:
Serverless cant have public IPs so requires a tunnel. We do not expose that in our connect workflow today.

## Screenshots (if appropriate)
Screen Shot 2021-02-02 at 11 21 19 AM

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rli — APPROVED** [review]

**Review by hunterwerlla — CHANGES_REQUESTED** [review]
Only my aurora database shows up with this change:
Screen Shot 2021-02-02 at 12 13 56 PM

**Comment by rli** [comment]
Might need to be a union of cluster and instance then?

**Comment by abrooksv** [comment]
> Might need to be a union of cluster and instance then?

Yep, looks like it.

File: jetbrains-ultimate/src-202+/software/aws/toolkits/jetbrains/services/rds/resources/RdsResources.kt [hunk 1]
```diff
@@ -25,5 +26,46 @@ val LIST_SUPPORTED_INSTANCES: Resource.Cached<List<DBInstance>> = ClientBackedCa
...
+                )
+            )
+        }.toList()
+}
+
+val LIST_SUPPORTED_CLUSTERS: Resource.Cached<List<RdsDatabase>> = ClientBackedCachedResource(RdsClient::class, "rds.list_supported_cluster") {
+    describeDBClustersPaginator {
+        it.filters(
+            Filter.builder()
+                .name(ENGINE_FILTER)
```
Comment (hunterwerlla) [inline jetbrains-ultimate/src-202+/software/aws/toolkits/jetbrains/services/rds/resources/RdsResources.kt:50]: Do we need a filter if we are selecting all?

File: jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/rds/RdsExplorerNodes.kt [hunk 2]
```diff
@@ -5,28 +5,33 @@ package software.aws.toolkits.jetbrains.services.rds
...
 import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerServiceRootNode
 import software.aws.toolkits.jetbrains.core.getResourceNow
+import software.aws.toolkits.jetbrains.core.utils.buildMap
+import software.aws.toolkits.jetbrains.services.rds.resources.LIST_SUPPORTED_CLUSTERS
 import software.aws.toolkits.jetbrains.services.rds.resources.LIST_SUPPORTED_INSTANCES
 
 class RdsExplorerParentNode(project: Project, service: AwsExplorerServiceNode) : AwsExplorerServiceRootNode(project, service) {
-    override fun getChildrenInternal(): List<AwsExplorerNode<*>> = nodeProject.getResourceNow(LIST_SUPPORTED_INSTANCES).map {
+    override fun getChildrenInternal(): List<AwsExplorerNode<*>> = buildMap<String, RdsDatabase> {
+        // De-dupe by db identifier
```
Comment (hunterwerlla) [inline jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/rds/RdsExplorerNodes.kt:19]: Add a comment that clusters take priority? Or does it matter? Is this the order the console does?

Comment (abrooksv) [reply jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/rds/RdsExplorerNodes.kt:19]: Console builds a tree, and puts instances under the cluster. (Show it shows everything)

File: jetbrains-ultimate/src-202+/software/aws/toolkits/jetbrains/services/rds/resources/RdsResources.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-ultimate/src-202+/software/aws/toolkits/jetbrains/services/rds/resources/RdsResources.kt:50]: We arent though, we dont have Mariadb, or Oracle

Comment (hunterwerlla) [reply jetbrains-ultimate/src-202+/software/aws/toolkits/jetbrains/services/rds/resources/RdsResources.kt:50]: Oh this is our `RdsEngine`. I was confused between our and the sdk's version of it

**Review by hunterwerlla — APPROVED** [review]
Works for me now

**Pull request merged**